relay-metrics/src/aggregator.rs

```diff
@@ -487,6 +487,23 @@ impl Aggregator {
         self.cost_tracker.totals_cost_exceeded(max_total_cost)
     }
 
+    /// Takes out the buckets from the aggregator, leaving an empty map.
+    pub fn take_buckets(&mut self) -> Vec<Bucket> {
```

Comment:
would have preffered if it took self by value, but couldn't please the borrow checker when doing the state change in `Project`